frontend/src/layouts/ProtectedLayout.vue

```diff
@@ -1,18 +1,55 @@
 <script setup lang="ts">
-import Header from "@/components/UI/Header.vue";
-import SideNav from "@/components/UI/SideNav.vue";
+import TermsAndConditions from "@/components/TermsAndConditions.vue";
+import Header from "@/layouts/Header.vue";
+import SideNav from "@/layouts/SideNav.vue";
+import Spinner from "@/components/UI/Spinner.vue";
+import { AppActlApiService } from "@/services/ApiServiceFactory";
+import { showTermsForAcceptance } from "@/store/TermsAndConditionsState";
+import { useQuery } from "@tanstack/vue-query";
+import { watch } from "vue";
+
+const termsAndConditionQuery = useQuery({
+    queryKey: ["user_terms_conditions", "user:validate"],
+    queryFn: () =>
+        AppActlApiService.userTermsAndConditionsApi
+            .validateUserRequiresAcceptTermsAndConditions()
+            .then((res) => res.data),
+});
+
+watch(termsAndConditionQuery.isSuccess, () => {
+    if (termsAndConditionQuery.data.value === true) {
+        showTermsForAcceptance();
+    }
+});
 </script>
 
 <template>
     <div>
         <!-- Show layout and content -->
+        <TermsAndConditions />
         <Header title="FAM" subtitle="Forests Access Management" />
         <SideNav />
         <div class="main">
             <main>
-                <router-view />
-                <!-- Render protected routes here -->
+                <div
+                    v-if="termsAndConditionQuery.isLoading.value"
+                    class="page-loading-spinner"
+                >
+                    <Spinner loading-text="Page loading" />
+                </div>
+                <router-view
+                    v-if="termsAndConditionQuery.data.value === false"
+                />
             </main>
         </div>
     </div>
 </template>
+<style lang="scss" scoped>
```

Comment:
Seems a weird place to define the spinner - could this be a component that's just referenced in one-line here? (There is a spinner component)

Reply:
https://github.com/bcgov/nr-forests-access-management/pull/1650/commits/5c42503b4cd7eb0fed00b3854d730752d2cac37b